ocp_resources/resource.py

```diff
@@ -461,25 +478,30 @@ def _base_body(self):
             if self.label:
                 self.res.setdefault("metadata", {}).setdefault("labels", {}).update(self.label)
 
-    def to_dict(self):
+    def to_dict(self) -> None:
         """
         Generate intended dict representation of the resource.
         """
         self._base_body()
 
-    def __enter__(self):
+    def __enter__(self) -> "Resource":
         signal(SIGINT, self._sigint_handler)
         return self.deploy()
 
-    def __exit__(self, exception_type, exception_value, traceback):
+    def __exit__(
+        self,
+        exc_type: type[BaseException] | None = None,
+        exc_val: BaseException | None = None,
+        exc_tb: TracebackType | None = None,
+    ) -> None:
         if self.teardown:
             self.clean_up()
 
-    def _sigint_handler(self, signal_received, frame):
-        self.__exit__(exception_type=None, exception_value=None, traceback=None)
+    def _sigint_handler(self, signal_received: int, frame: Any) -> None:
+        self.__exit__()
         sys.exit(signal_received)
 
-    def deploy(self, wait=False):
+    def deploy(self, wait: bool = False) -> "Resource":
```

Comment:
Is this `Resource` string or ResourceInstance?

Reply:
should be `"Resource"`, which mean `self`